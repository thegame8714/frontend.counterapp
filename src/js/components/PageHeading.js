import React, { Component } from "react";
import _ from 'lodash';

class PageHeading extends Component {

  /**
   * 
   * 
   * @returns
   * 
   * @memberOf PageHeading
   * use lodash to assign fallback value in case issues with react router
   */
  render() {
    return (
      <h1>{_.get(this.props.route,
        this.props.pageTitle,
        this.props.value)}</h1>
    );
  }
}

export default PageHeading;
