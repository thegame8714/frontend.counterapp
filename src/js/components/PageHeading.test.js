import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import PageHeading from './PageHeading';

describe('PageHeading', () => {
  it('should render page tile as H1 with text', () => {
    const route = {
      object: "this.props.route",
      path: "pageTitle",
      value: "Heading Value"
    }
    const wrapper = mount(<PageHeading route={route} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});