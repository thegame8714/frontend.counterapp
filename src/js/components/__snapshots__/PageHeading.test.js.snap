exports[`PageHeading should render page tile as H1 with text 1`] = `
<PageHeading
  route={
    Object {
      "object": "this.props.route",
      "path": "pageTitle",
      "value": "Heading Value",
    }
  }>
  <h1 />
</PageHeading>
`;
